File: 48 Rotate Image.py
#這題python先做出來
#負的索引值真是太方便看了
class Solution:
    def rotate(self, matrix):
        l = len(matrix)
        for i in range(int(l / 2)):
            for j in range(i, l-i-1):
                temp = matrix[i][j]
                matrix[i][j] = matrix[-j-1][i]
                matrix[-j-1][i] = matrix[-i-1][-j-1]
                matrix[-i-1][-j-1] = matrix[j][-i-1]
                matrix[j][-i-1] = temp

File: 48 Rotate Image.cpp
class Solution {
public:
    void rotate(vector<vector<int>>& matrix) {
        int temp;
        int length = matrix.size();
        for(int i = 0;i < length / 2;++i){
            for(int j = i;j < length-i-1;++j){
                temp = matrix[i][j];
                matrix[i][j] = matrix[length-j-1][i];
                matrix[length-j-1][i] = matrix[length-i-1][length-j-1];
                matrix[length-i-1][length-j-1] = matrix[j][length-i-1];
                matrix[j][length-i-1] = temp;
            }
        }
    }
};